x-pack/plugins/triggers_actions_ui/public/application/sections/rules_list/components/notify_badge/types.ts

```diff
@@ -7,18 +7,26 @@
 
 import { RuleTableItem, SnoozeSchedule } from '../../../../../types';
 
+export type RuleSnoozeSettings = Pick<
+  RuleTableItem,
+  'id' | 'activeSnoozes' | 'isSnoozedUntil' | 'muteAll' | 'snoozeSchedule'
+>;
+
 export interface RulesListNotifyBadgeProps {
-  rule: Pick<
-    RuleTableItem,
-    'id' | 'activeSnoozes' | 'isSnoozedUntil' | 'muteAll' | 'isEditable' | 'snoozeSchedule'
-  >;
-  isOpen: boolean;
-  isLoading: boolean;
-  previousSnoozeInterval?: string | null;
-  onClick: React.MouseEventHandler<HTMLButtonElement>;
-  onClose: () => void;
-  onLoading: (isLoading: boolean) => void;
-  onRuleChanged: () => void;
+  /**
+   *  Rule's snooze settings, if `undefined` is passed the component is shown in loading state
+   */
+  snoozeSettings: RuleSnoozeSettings | undefined;
```

Comment:
@Zacqary I see your point. But the reason behind renaming is a bit deeper. There is a well known [SOLID](https://en.wikipedia.org/wiki/SOLID) principle in software engineering. So I'd like to achieve interface segregation here. Rule snooze functionality is just one feature among the others and decoupling it will help later on. So far rule snooze settings represent a set of fields [spread across the rule object](https://github.com/elastic/kibana/blob/main/x-pack/plugins/alerting/common/rule.ts#L159). In ideal case we shouldn't mix how we store the data and business model representation in the other words rule snooze fields could be stored in one SO as spread fields or under some key, or in a side car SO, or in any other storage but it shouldn't affect the business logic. So `Rule` type could be a union of interfaces so there is no hard to feed `rule` object to the component then.

In Security Solution we also decided to follow interface segregation principle and [fetch rule snooze setting via API](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/detection_engine/rule_management/api/api.ts#L198-L224) so only `id`, `muteAll`, `snoozeSchedule`, `activeSnoozes` and `isSnoozedUntil` fields. This way we achieve decoupling and it's a bit weird to name such an object containing only these 5 fields a rule.

Reply:
Let's separate the rule `id` from the `snoozeSettings` object then, as it's only needed for `RulesListNotifyBadgeWithApi`. `id` is never not going to be a rule `id` so that's the big thing giving me pause on calling the whole structure `snoozeSettings`.